Ansible for Control Node comes with distribution (#4027)
uyuni-project/uyuni-docs: modules/administration/pages/ansible-setup-control-node.adoc

To set up an Ansible control node, execute the following steps from the {productname} {webui}.

[NOTE]
====
To configure a {productname} client the Ansible software must be available for operating system of the client.
This usually means Ansible must come from the client's software distribution.
For example, Ansible is available for {sle} 15 SP6 or SP7.

> ====

Reviewer: I added a suggestion but feel free to modify it

====

> . In the {productname} {webui}, navigate to menu:Admin[Setup Wizard > Products], verify that [literal]``SUSE Linux Enterprise Server 15 SP5 x86_64`` and [literal]``{productname} Client Tools for SLE 15 x86_64`` are selected and synchronized.

Reviewer 2: Shouldn't we mention that the old ansible version can be found in SLE15 client tools? We are dropping all the information about this in the page

Reviewer: We have new set of SLE15 client tools in 5.1 and those client tools don't have ansible. 

For 5.0, yes, we need to keep the info about the client tools.

Reviewer 2: Hmmm, I still see `ansible` at:
- https://build.suse.de/package/show/Devel:Galaxy:Manager:Head:MLMTools-Beta-SLE15/ansible
- https://build.suse.de/package/show/Devel:Galaxy:Manager:5.1:MLMTools-SLE15/ansible

Aren't these the new client tools? Maybe the package are present even if not really submitted.. we would need some clarifications.

Author: I'll merge this one now.  If needed we must create a follow-up PR.

Reviewer: @meaksh sorry missed your comment before. I don't know why ansible package is displayed in IBS but I don't see that here https://updates.suse.de/SUSE/Products/MultiLinuxManagerTools/SLE-15/x86_64/product/ and it isn't expected. I will check with @deneb-alpha

. In the {productname} {webui}, navigate to menu:Admin[Setup Wizard > Products], verify that [literal]``SUSE Linux Enterprise Server 15 SP6 x86_64`` (or later) with the [literal]``Systems Management Module`` and the required [literal]``Python 3 Module`` are selected and synchronized.

. Deploy a {sle} 15 SP6 (or later) client.

. In the {productname} {webui}, navigate to the menu:Systems[Overview] page of the client.

>   Select menu:Software[Software Channels] and subscribe the client to the [literal]``SUSE Linux Enterprise Server 15 SP5 x86_64``, [literal]``Systems Management Module`` and  [literal]``Python 3 Module`` channels.

Reviewer 2: I think there is a typo here, I would suggest:
```suggestion
  Select menu:Software[Software Channels] and subscribe the client to the [literal]``SUSE Linux Enterprise Server 15 SP6 x86_64`` (or later SP), [literal]``Systems Management Module`` and  [literal]``Python 3 Module`` channels.
```

To set up an Ansible control node, execute the following steps from the {productname} {webui}.

[NOTE]
====

> The following procedure applies for {sle} 15 system only.

Reviewer: We are not shipping ansible in any of our client tools anymore in 5.1. So we should adjust it accordingly. For every distribution, ansible needs to come from the distrubution.

For 15SP6 and SP7, it should be coming be coming from `Systems Management Module` which has dependency on `Python 3 Module` so both of these modules needs to be enabled if user wants to install ansible package on SP6/SP7 system in order to make them control node.

For SP4 and SP5, if user want to install new Ansible, they need to enable only Python 3 Module ` , this is what that bug was about where you were tagged.

We also need to add a note with newer ansible, systems with old python cannot be managed anymore.

Author: Thanks.  I added all this. I think here we shouldn't document Ansible version related limitations that detailed.  Focus here should be setting up a (new) Ansible control with MLM 5.1.

[NOTE]
====

> On {sle} 15, the {productname} client tools contain the [package]``ansible`` package.

Reviewer: This needs to be removed or modified with what I have described above.

. Select the menu:Events[] tab and verify the status of the highstate.

[NOTE]
====

> If you want to install a newer Ansible on a {sle} 15 SP4 or SP5 client, you must enable only the [literal]``Python 3 Module``.

Reviewer: remove the 'only'

Author: ok

> If you want to install a newer Ansible on a {sle} 15 SP4 or SP5 client, you must enable only the [literal]``Python 3 Module``.

Author: ```suggestion
If you want to install a newer Ansible on a {sle} 15 SP4 or SP5 client, you must enable the [literal]``Python 3 Module``.
```

====

[NOTE]
====
With newer Ansible, systems with an old Python version cannot be managed anymore.

> ====

Reviewer: We can probably extend it with something like this 

Newer versions of Ansible no longer support managing nodes with outdated Python versions 
If a managed node still defaults to an older Python version, you may encounter connection errors or failures during playbook runs. To address this, user should  Upgrade Python on the managed node, if possible and Set the correct Python interpreter in your Ansible inventory or configuration.

Reviewer: @meaksh can you please verify this

Reviewer 2: that's correct :+1:

. Select the menu:Events[] tab and verify the status of the highstate.

[NOTE]
====
If you want to install a newer Ansible on a {sle} 15 SP4 or SP5 client, you must enable the [literal]``Python 3 Module``.
====

[NOTE]
====
With newer Ansible, systems with an old Python version cannot be managed anymore.
====

> //Procedure: Installing the SCAP Security Guide Package
> include::snippets/install_scap_security_guide_package.adoc[]

Author: ```suggestion
```
